data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -0,0 +1,253 @@
+            return;
+        } else {
+            String faraoContingencyId = importedCsaProfileContingencyCreationContextOpt.get().getContigencyId();
+            Optional<String> normalEnabledOpt = Optional.ofNullable(contingencyWithRemedialActionPropertyBag.get(CsaProfileConstants.CONTINGENCY_WITH_REMEDIAL_ACTION_NORMAL_ENABLED));
+            if (normalEnabledOpt.isPresent() && !Boolean.parseBoolean(normalEnabledOpt.get())) {
+                csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "normalEnabled must be true or empty"));
+            } else {
+                Set<NetworkElement> networkElements = crac.getContingency(contingencyId).getNetworkElements();
+                if (networkElements.isEmpty()) {
+                    csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "Remedial action" + remedialActionId + "has to have at least one ElementaryAction"));
+                    return;
+                }
+            }
+            faraoContingenciesIds.add(faraoContingencyId);
+        }
+    }
+
+    private void importOnInstant(NetworkActionAdder networkActionAdder, PropertyBag topologyActionPropertyBag, String remedialActionId, List<String> elementaryActions) {
+        String switchId = topologyActionPropertyBag.getId(CsaProfileConstants.SWITCH);
+        if (network.getSwitch(switchId) == null) {
+            csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "No switch with id: " + switchId + " found in network"));
+            return;
+        }
+        String propertyReference = topologyActionPropertyBag.getId(CsaProfileConstants.GRID_ALTERATION_PROPERTY_REFERENCE);
+        if (!propertyReference.equals(CsaProfileConstants.PROPERTY_REFERENCE_SWITCH_OPEN)) {
+            // todo this is a temporary behaviour closing switch will be implemented in a later version
+            csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "Only Switch.open propertyReference is supported"));
+            return;
+        }
+
+        networkActionAdder.newTopologicalAction()
+                .withNetworkElement(switchId)
+                .withActionType(ActionType.OPEN).add();
+        elementaryActions.add(switchId);
+    }
+
+    private Optional<Integer> getSpeedOpt(String timeToImplement) {
+        if (timeToImplement != null) {
+            return Optional.of(CsaProfileCracUtils.convertDurationToSeconds(timeToImplement));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private Optional<String> createRemedialActionName(String nativeRemedialActionName, String tsoName) {
```

Comment:
Was not sure if it can do the same thing (it doesn't handle null values ) , but in any case i will let the conversation unresolved, and we'll discuss later how we can put things in common